File: java/src/main/java/de/oglimmer/mastermind/Color.java
package de.oglimmer.mastermind;

import java.util.Arrays;

public enum Color {

    RED('r'),
    BLUE('b'),
    GREEN('g'),
    YELLOW('y'),
    ORANGE('o'),
    WHITE('w');

    Color(char key) {
        this.key = key;
    }

    private char key;

    public static Color getRandom() {
        int rnd = (int) (Math.random() * 6);
        return Color.values()[rnd];
    }

    public static boolean isValid(char c) {
        return Arrays.stream(values()).anyMatch(e -> e.key == c);
    }

    public static Color get(Character character) {
        return Arrays.stream(Color.values()).filter(e -> e.key == character).findAny().get();
    }

    public String toString() {
        return Character.toString(key);
    }
}

File: swift/Sources/MasterMind/Color.swift
//
//  Color.swift
//  MasterMind
//
//  Created by oglimmer on 11/12/21.
//

import Foundation

class Color : CustomStringConvertible {
    static var RED = Color(identifier: "r")
    static var BLUE = Color(identifier: "b")
    static var GREEN = Color(identifier: "g")
    static var YELLOW = Color(identifier: "y")
    static var ORANGE = Color(identifier: "o")
    static var WHITE = Color(identifier: "w")
    static var ALL = [RED, BLUE, GREEN, YELLOW, ORANGE, WHITE]

    init(identifier: Character) {
        self.identifier = identifier
    }
    
    let identifier: Character
    
    static func getRandom() -> Color {
        let rnd = Int.random(in: 0..<6)
        return ALL[rnd];
    }

    static func isValid(c: Character) -> Bool {
        return ALL.contains(where: { $0.identifier == c })
    }

    static func get(character: Character) -> Color {
        return ALL.first(where: { $0.identifier == character })!
    }
    
    public var description: String {
        return String(identifier)
    }
}
